src/v/cloud_storage/tx_range_manifest.cc

```diff
@@ -47,10 +243,34 @@ ss::future<> tx_range_manifest::update(ss::input_stream<char> is) {
     });
     iobuf_istreambuf ibuf(result);
     std::istream stream(&ibuf);
-    Document m;
     IStreamWrapper wrapper(stream);
-    m.ParseStream(wrapper);
-    do_update(m);
+
+    switch (_deserializer) {
+    case deserializer::rapidjson_dom: {
+        Document m;
+        m.ParseStream(wrapper);
+        do_update(m);
+        break;
+    }
+    case deserializer::rapidjson_sax: {
+        rapidjson::Reader reader;
+        tx_range_manifest_json_handler handler;
+
+        if (reader.Parse(wrapper, handler)) {
```

Comment:
I think you should check 'fail' bit. In case of DOM this is not as important because you will get different error if the stream is truncated for some reason. But in case of SAX parser the parser will just stop calling methods of the handler and will return incomplete result. There is no check that checks that the `handler._state == expect_done` or something like that.